djstripe/models.py

```diff
@@ -457,94 +525,124 @@ def sync_charges(self, cu=None, **kwargs):
         cu = cu or self.stripe_customer
         for charge in cu.charges(**kwargs).data:
             self.record_charge(charge.id)
-
-    def sync_current_subscription(self, cu=None):
+            
+    def sync_subscriptions(self, cu=None):
+        """
+        Remove all existing Subscription records and regenerate from the Stripe
+        subscriptions.
+        """
         cu = cu or self.stripe_customer
-        sub = cu.subscription
-        if sub:
-            try:
-                sub_obj = self.current_subscription
-                sub_obj.plan = djstripe_settings.plan_from_stripe_id(sub.plan.id)
-                sub_obj.current_period_start = convert_tstamp(
-                    sub.current_period_start
-                )
-                sub_obj.current_period_end = convert_tstamp(
-                    sub.current_period_end
-                )
-                sub_obj.amount = (sub.plan.amount / decimal.Decimal("100"))
-                sub_obj.status = sub.status
-                sub_obj.cancel_at_period_end = sub.cancel_at_period_end
-                sub_obj.canceled_at = convert_tstamp(sub, "canceled_at")
-                sub_obj.start = convert_tstamp(sub.start)
-                sub_obj.quantity = sub.quantity
-                sub_obj.save()
-            except CurrentSubscription.DoesNotExist:
-                sub_obj = CurrentSubscription.objects.create(
-                    customer=self,
-                    plan=djstripe_settings.plan_from_stripe_id(sub.plan.id),
-                    current_period_start=convert_tstamp(
+        subs = cu.subscriptions
+        if Customer.allow_multiple_subscriptions and not Customer.retain_canceled_subscriptions:
+            self.subscriptions.all().delete()
+        if subs.count > 0:
+            for sub in subs.data:
+                try:
+                    if Customer.allow_multiple_subscriptions:
+                        sub_obj = self.subscriptions.get(stripe_id=sub.id)
```

Comment:
And make similar changes elsewhere, for example in subscribe() and update_plan_quantity().